File: leetcode/leetcode_py/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        m = {}
        for i, n in enumerate(nums):
            if n in m:
                if abs(m[n] - i) <= k:
                    return True
            m[n] = i
        return False
        

File: leetcode/leetcode_cpp/contains-duplicate-ii.cpp
/*
https://leetcode.com/problems/contains-duplicate-ii

Solutions. Hash map
time: o(n)
space: o(n)

*/
class Solution {a
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> m;
        for (int i=0; i<nums.size(); ++i) {
            if (m.count(nums[i])) {
                if (abs(m[nums[i]] - i) <= k) return true;    
            }
            m[nums[i]] = i;
        }
        return false;
    }
};